docs/workers.md

```diff
@@ -147,7 +147,9 @@ In the config file for each worker, you must specify:
    the main process (`worker_main_http_uri`). This config option is no longer required and is ignored when running Synapse 1.73 and newer.
  * **Synapse 1.83 and older:** The HTTP replication endpoint that the worker should talk to on the main synapse process
    ([`worker_replication_host`](usage/configuration/config_documentation.md#worker_replication_host) and
-   [`worker_replication_http_port`](usage/configuration/config_documentation.md#worker_replication_http_port)). If using Synapse 1.84 and newer, these are not needed if `main` is defined on the [shared configuration](#shared-configuration) `instance_map`
+   [`worker_replication_http_port`](usage/configuration/config_documentation.md#worker_replication_http_port)). If
```

Comment:
I would probably just have nuked those from this doc. Upgrade guide is wordy and good enough I feel.

Reply:
yeah, we should remove the removed options from the manual. They are still available in the old versions of the manual for people who really need them.

Reply:
I'm onboard, but want to make sure not to just take an eraser to the whole thing yet, as there will invariably be someone who misses checking upgrade notes.

There are a couple places where this is mentioned:
1. workers.md:
   * This spot, actually. The brief summary of what you need to get a worker up and running.
2. config_documentation.md
   * the [itemized list](https://github.com/matrix-org/synapse/blob/39d131b016673bbd4d3c28095c8838b8c6dc0953/docs/usage/configuration/config_documentation.md?plain=1#L4093-L4136) of what the settings are

So, number 2 I'm good with losing now. And I can update the links in 1 to point at the old manual. Is that what you mean?

Reply:
OMG yeah the itemized list too... Please nuke those for sure, personally I would also nuke legacy from workers doc but I'll survive if it stays there :) 

I think legacy should be dealt with the upgrade notes, with a warning when we encounter an unknown option to check the notes. But we can do that later.

Reply:
There. I just took it all out. Since it is preserved in old versions of the manual and is on the upgrade notes, I guess that's it?

Removed in [a99cdac](https://github.com/matrix-org/synapse/pull/15860/commits/a99cdac85137aa34440f2bf7569df7a1e217e1da)